src/code/z_bgcheck.c

```diff
@@ -4112,14 +4112,15 @@ s32 SurfaceType_IsIgnoredByProjectiles(CollisionContext* colCtx, CollisionPoly*
 }
 
 /**
- * CollisionPoly is conveyor enabled
- * Returns true if `poly` is a conveyor surface, else false
+ * SurfaceType Get Conveyor Surface Type
+ * Return type 0 (CONVEYOR_WATER) if 'poly' is a surface that will only move player underwater
+ * Return type 1 (CONVEYOR_FLOOR) if `poly` is a surface that must be stood on to move player
  */
-s32 SurfaceType_IsConveyor(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
+s32 SurfaceType_GetConveyorType(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
```

Comment:
This again might be a bad example since it comes from MM, but the motivation for thinking of it as a type was because the way it was previously, there was a function:
```
if (!SurfaceType_IsConveyor(&sQuake2Context.globalCtx->colCtx, player->actor.floorPoly, player->actor.floorBgId)) {
    return SurfaceType_GetConveyorSpeed(&sQuake2Context.globalCtx->colCtx, player->actor.floorPoly, player->actor.floorBgId);
}
```
(i.e. only return a conveyor value if the surface is not a conveyor)

That made it seem like it was looking for a specific type. But you're right in that the double `!!` is usually for a bool. So I couldn't decide between the two. Based on the code from player though, it did seem like something specific was going on for each case:
- returning a 0 needs to be paired with `(this->actor.yDistToWater > 20.0f)`
- returning a 1 needs to be paired with `(this->actor.bgCheckFlags & 1)`

I might be fine with something like `IsConveyorOnLand` though

Reply:
To add to Dragorn's points, consider that any surface that isn't intended to be a conveyor must return 0. Polys that aren't beneath water are far more common than ones that are, and so it becomes weird to say that these surfaces are water conveyors. 

I would rather go with something to the effect of `IsFloorConveyor`, since calling it "Land" would introduce new terminology.

Reply:
It seems to me that any surface that isn't intended to be a conveyor is characterized to not be a conveyor by having a speed of 0, instead of having the function name in question return 0. That's why it's the top level function called, and checks that it's not 0:
https://github.com/zeldaret/oot/blob/92bb50e2966b820db00c627ee71e0aaee897ef31/src/overlays/actors/ovl_player_actor/z_player.c#L9461

I can't find a single instance in either OoT or MM codebase where `SurfaceType_IsConveyor == 0` is used to define a surface as not being conveyor.

That said, if we still wanted to go with the boolean approach, I do agree with using `Floor`, not `Land` would be better

Reply:
Actually, thinking about it more, I think I see your point. I still think that a speed of 0 is a better indicator of not being a conveyor, as then `SurfaceType_GetConveyorType` doesn't matter in the end, but for non-water surfaces, setting it to 0 is another way to make a surface not a conveyor.

i.e. by setting the speed to 0, you ensure the surface is not a conveyor. However, alternatively, if you set what I'm calling the type to 0, it only ensures non-water based surfaces are not conveyors, i.e. only a subset of surfaces that don't interact with water are not conveyors. Which is a less powerful statement than setting the speed to 0. Does that make sense?

Reply:
How I see it is that `SurfaceType_GetConveyorSpeed` returning 0 means not a conveyor
If a conveyor, `SurfaceType_GetConveyorType` is then used to check in which case it is supposed to apply to the player.

I agree with the idea that `SurfaceType_GetConveyorType` should be a "type", but evidence (`!!` and checks against 0 only) shows it's meant as a boolean

Reply:
I'd be fine with `SurfaceType_IsConveyorFloorType` and treating it as a bool then. Or some variation similar to that